src/client/views/book/bookEditorInterface.ml

```diff
@@ -109,25 +130,19 @@ let create ?on_save page =
 
   let sets_area = Html.createDiv (Page.document page) in
   let sets_search =
-    let main_section =
-      SearchBar.Section.create
-        ~search:(fun input ->
-            let%rlwt formula = Lwt.return @@ Result.map_error List.singleton @@ Set.Filter.from_string input in
-            let%lwt results =
-              Set.search ~threshold:0.4
-                ~pagination:Pagination.{start = 0; end_ = 10} formula
-            in
-            Lwt.return_ok results)
-        ~make_result:(fun score -> make_set_search_result editor page score)
-        page
-    in
-    SearchBar.create
-      ~placeholder:"Add set (Magic Search)"
-      ~sections:[main_section]
-      page
+    let open Dancelor_client_html in
+    To_dom.of_div @@ div [
+      Dancelor_client_components.SearchBar.make
+        ~placeholder:"Add set (Magic Search)"
+        ~search
+        ~make_result:(make_set_result editor page)
+        ~max_results:10
+        ~on_enter:(fun search_text ->
+            Dom_html.window##.location##.href := js PageRouter.(path (Search (Some search_text))))
```

Comment:
This last `on_enter` will trigger a redirection to the search page if we press enter. Is that really what we want? I feel we'd rather just not have an action on enter at all (and no “Press enter for more results” message from the bar). Is that correct or am I missing something?

Reply:
Very good point, done in [3a41135](https://github.com/paris-branch/dancelor/pull/252/commits/3a4113543192f65b873c0d8e1bb0cdc17f144950)